src/hardware/openbench-logic-sniffer/api.c

```diff
@@ -411,10 +411,8 @@ static int dev_acquisition_start(const struct sr_dev_inst *sdi)
 	/* If the device stops sending for longer than it takes to send a byte,
 	 * that means it's finished. But wait at least 100 ms to be safe.
 	 */
-	serial_source_add(sdi->session, serial, G_IO_IN, 100,
+	return serial_source_add(sdi->session, serial, G_IO_IN, 100,
```

Comment:
I was thinking if this could regress somehow, but I think this is just better :)